pkg/controller/master/setting/register.go

```diff
@@ -74,24 +80,91 @@ func Register(ctx context.Context, management *config.Management, options config
 	}
 
 	syncers = map[string]syncerFunc{
-		"additional-ca":             controller.syncAdditionalTrustedCAs,
-		"cluster-registration-url":  controller.registerCluster,
-		"http-proxy":                controller.syncHTTPProxy,
-		"log-level":                 controller.setLogLevel,
-		"overcommit-config":         controller.syncOvercommitConfig,
-		"vip-pools":                 controller.syncVipPoolsConfig,
-		"auto-disk-provision-paths": controller.syncNDMAutoProvisionPaths,
-		"ssl-certificates":          controller.syncSSLCertificate,
-		"ssl-parameters":            controller.syncSSLParameters,
-		"containerd-registry":       controller.syncContainerdRegistry,
-		"ntp-servers":               controller.syncNTPServer,
-		"auto-rotate-rke2-certs":    controller.syncAutoRotateRKE2Certs,
+		"additional-ca":                              controller.syncAdditionalTrustedCAs,
+		"cluster-registration-url":                   controller.registerCluster,
+		"http-proxy":                                 controller.syncHTTPProxy,
+		"log-level":                                  controller.setLogLevel,
+		"overcommit-config":                          controller.syncOvercommitConfig,
+		"vip-pools":                                  controller.syncVipPoolsConfig,
+		"auto-disk-provision-paths":                  controller.syncNDMAutoProvisionPaths,
+		"ssl-certificates":                           controller.syncSSLCertificate,
+		"ssl-parameters":                             controller.syncSSLParameters,
+		"containerd-registry":                        controller.syncContainerdRegistry,
+		harvSettings.NTPServersSettingName:           controller.syncNodeConfig,
+		harvSettings.LonghornV2DataEngineSettingName: controller.syncNodeConfig,
+		"auto-rotate-rke2-certs":                     controller.syncAutoRotateRKE2Certs,
 		harvSettings.KubeconfigDefaultTokenTTLMinutesSettingName: controller.syncKubeconfigTTL,
 		// for "backup-target" syncer, please check harvester-backup-target-controller
 		// for "storage-network" syncer, please check harvester-storage-network-controller
 	}
 
 	settings.OnChange(ctx, controllerName, controller.settingOnChanged)
 	apps.OnChange(ctx, controllerName, controller.appOnChanged)
+	node.OnChange(ctx, controllerName, controller.nodeOnChanged)
 	return nil
 }
+
+func (h *Handler) nodeOnChanged(_ string, node *corev1.Node) (*corev1.Node, error) {
+	if node == nil || node.DeletionTimestamp != nil {
+		return node, nil
+	}
+
+	// This ensures newly added cluster nodes have a node config created
+	// which is necessary to ensure the v2 data engine setting is correct
+	// on newly added nodes.
+	nodeConfig, err := h.getOrCreateNodeConfigForNode(node)
```

Comment:
the nodeConfig is not really needed until we have figured out if we need to enable/disable v2 engine, L148. 

may be we could move this check there and avoid having to runs this method and ops unless they are needed

Reply:
There's one annoying case that we need this line here for, which is: when someone has already enabled the v2 engine, and then later they add a new node.  That node won't have any labels set (the label is only for if you want to _disable_ the engine on a node), so the code would never get past line 128 in this function.

Or, is there another way I can catch the "new node was added" case, without doing this on every node change?

Reply:
Oh, hey, what about this? When we create the node config the first time inside `getOrCreateNodeConfigForNode()`, we add an annotation to the node (something like "node.harvesterhci.io/node-config-created=true"), then in `nodeOnChanged()` we check for that annotation before the above call. That way in the normal case the only thing that executes is a quick lookup in the node's annotations map.